app/config/Switches.scala

```diff
@@ -0,0 +1,103 @@
+package config
+
+import akka.agent.Agent
+import com.amazonaws.services.s3.model.{GetObjectRequest, ObjectMetadata, PutObjectRequest, PutObjectResult}
+import com.amazonaws.util.StringInputStream
+import org.quartz._
+import org.quartz.impl.StdSchedulerFactory
+import play.api.Logger
+import play.api.libs.json.{Format, Json}
+import play.json.extra.Variants
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.Source
+
+object Switches {
+  def allSwitches: Map[String, SwitchState] = agent.get()
+  private val agent = Agent[Map[String, SwitchState]](Map.empty)
+  private val bucket = "login-gutools-config"
+  private val scheduler = StdSchedulerFactory.getDefaultScheduler
+  val loginConfig = LoginConfig.loginConfig(AWS.eC2Client)
+  val fileName = s"${loginConfig.stage.toUpperCase}/switches.json"
+
+  class SwitchJob extends Job() {
+    override def execute(context: JobExecutionContext): Unit = refresh()
+  }
+  private val job = JobBuilder.newJob(classOf[SwitchJob])
+    .withIdentity("refresh-switches-gu-login-tools")
+    .build()
+
+  def setEmergencySwitch(state: SwitchState): Option[PutObjectResult] = {
```

Comment:
Since `PutObjectResult` is an internal AWS type that doesn't really have anything to do with switches, would it be better to return something that has more to do with what's being done here?

Unit would make sense since this is explicitly a side-effect function.